app/parse.py

```diff
@@ -2,11 +1,156 @@
+import csv
+import time
+
+import requests
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from dataclasses import dataclass, fields
+from bs4 import BeautifulSoup
+
+BASE_URL = "https://mate.academy"
+
+
+class Driver:
+    instance = None
+
+    @staticmethod
+    def initialize() -> webdriver.Chrome:
+        if Driver.instance is None:
+            Driver.instance = webdriver.Chrome()
```

Comment:
Secondly, this method is not thread-safe. If multiple threads call this method concurrently, it's possible that multiple WebDriver instances will be created. Consider using a thread-safe singleton pattern if you expect this code to be used in a multi-threaded environment.